From 8b16a5e3958ecdd40b0599c57516fbae0b65343d Mon Sep 17 00:00:00 2001
From: Enrico Jorns <ejo@pengutronix.de>
Date: Sun, 25 Jun 2023 22:14:21 +0200
Subject: [PATCH 2/2] script: add nfs-cp helper

This allows copying files into the nfs export directory.

E.g. to copy a new fstab to your nfs-exported rootfs at
tmp/deploy/nfsroot, run:

  $ nfs-cp tmp/deploy/nfsroot fstab /etc/fstab

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 scripts/nfs-cp | 49 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)
 create mode 100755 scripts/nfs-cp

diff --git a/scripts/nfs-cp b/scripts/nfs-cp
new file mode 100755
index 0000000000..f9b6932bb5
--- /dev/null
+++ b/scripts/nfs-cp
@@ -0,0 +1,49 @@
+#!/bin/bash
+#
+# This utility copies a file into an exported rootfs using pseudo
+#
+# Copyright (c) 2023 Pengutronix
+#
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+function usage() {
+	echo "Usage: $0 <extract-dir> <source> <dest>"
+}
+
+if [ $# -ne 3 ]; then
+	usage
+	exit 1
+fi
+
+SYSROOT_SETUP_SCRIPT=`which oe-find-native-sysroot 2> /dev/null`
+if [ -z "$SYSROOT_SETUP_SCRIPT" ]; then
+	echo "Error: Unable to find the oe-find-native-sysroot script"
+	echo "Did you forget to source your build system environment setup script?"
+	exit 1
+fi
+. $SYSROOT_SETUP_SCRIPT qemu-helper-native
+PSEUDO_OPTS="-P $OECORE_NATIVE_SYSROOT/usr"
+
+SDK_ROOTFS_DIR=$1
+SOURCE_PATH=$2
+DEST_PATH=$3
+
+# Convert SDK_ROOTFS_DIR to a full pathname
+if [[ ${SDK_ROOTFS_DIR:0:1} != "/" ]]; then
+	SDK_ROOTFS_DIR=$(readlink -f $(pwd)/$SDK_ROOTFS_DIR)
+fi
+
+pseudo_state_dir="$SDK_ROOTFS_DIR/../$(basename "$SDK_ROOTFS_DIR").pseudo_state"
+pseudo_state_dir="$(readlink -f $pseudo_state_dir)"
+
+echo "PSEUDO_LOCALSTATEDIR: $pseudo_state_dir"
+
+PSEUDO_LOCALSTATEDIR="$pseudo_state_dir"
+export PSEUDO_LOCALSTATEDIR
+
+echo "Copying file using pseudo..."
+echo "$PSEUDO $PSEUDO_OPTS cp $SOURCE_PATH $SDK_ROOTFS_DIR/$DEST_PATH"
+$PSEUDO $PSEUDO_OPTS cp $SOURCE_PATH $SDK_ROOTFS_DIR/$DEST_PATH
+
+exit 0
-- 
2.43.0

